### PR TITLE
Gateway followup for 5.2.2

### PR DESCRIPTION
**Pull request by dominikl**
Java Gateway extensions targeted for 5.2.2 ( cherry-picked from #4347 )

The commits already have been reviewed in above mentioned PR, but I think there should be some kind of (hopefully) final review. The only "controversial" thing left in this PR is the usage of an `ExecutorService` for asynchronous method calls (for now this is only used for file attachment uploads).

### CODE REVIEW

**Comment by jburel** [comment]
I will probably keep the following constructor:
```public Gateway(Logger log, CacheService cacheService)```

**Comment by jburel** [comment]
Do we want to put the ```uploadImage``` this round?

**Comment by jburel** [comment]
I cannot find tests for ```getOrphanedImages``` and ```findObject``` methods

**Comment by dominikl** [comment]
Ah, I guess the problem is the whole bioformats dependency then, isn't it? I'll remove the `uploadImage` method for now; and check/implement tests for `getOrphandedImages` and `findObject`, thanks.

File: components/blitz/src/omero/gateway/facility/ROIFacility.java [hunk 1]
```diff
@@ -43,7 +43,6 @@
 import omero.gateway.model.ROIResult;
 import omero.gateway.util.ModelMapper;
 import omero.gateway.util.PyTablesUtils;
-import omero.model.IObject;
```
Comment (sbesson) [inline components/blitz/src/omero/gateway/facility/ROIFacility.java]: @dominikl: can we leave this `import` cleanup out of this PR? As far as I can tell, this is the only place that would remain in conflict with `regions`.

Comment (dominikl) [inline components/blitz/src/omero/gateway/facility/ROIFacility.java]: Yes, I'll restore the imports to match the `regions` ones. I just noticed that I also should remove the `PyTablesUtils` and `TableData` classes. Not sure how they slipped in there, these classes should have been part of the `TablesFacility` commit, which I did not cherry-pick for this PR.

**Comment by dominikl** [comment]
Forgot the Java examples, training job failed. Will have to fix that in this PR, too.

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/OMERO-DEV/job/OMERO-DEV-merge-training/246/ is green 
thanks @dominikl

**Comment by jburel** [comment]
```Executors.newCachedThreadPool()```. Was there some performance/cpu usage testing for long lasting tasks?
The upload files is probably short enough that no impact was noticeable

**Comment by dominikl** [comment]
No... didn't do any performance test. What would be a good test scenario?

**Comment by jburel** [comment]
Uploading simultaneously some large attachments will probably be enough for now.

**Comment by dominikl** [comment]
Looks good, the upload time per file is even shorter with a multiple parallel upload compared to a single file attachment upload. I added an integration test for that, see last commit.

**Comment by jburel** [comment]
@dominikl: thanks for adding the test.

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/OMERO-DEV/job/OMERO-DEV-merge-integration-java/214/
new test did not finish within the timeout set.

**Comment by jburel** [comment]
Transient failure https://ci.openmicroscopy.org/view/OMERO-DEV/job/OMERO-DEV-merge-integration-java/215/testngreports/integration.delete/DeleteProjectedImageTest/testBothImagesByOwnerdeleteByOwnerRWR/ (timeout)
not related to this PR.
Thanks @dominikl for reworking it and adding the tests.

File: components/tools/OmeroJava/test/integration/gateway/SearchFacilityTest.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ protected void setUp() throws Exception {
         super.setUp();
         initData();
         // wait a little bit for the search indexer
-        Thread.sleep(10000);
+        Thread.sleep(30000);
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/gateway/SearchFacilityTest.java:74]: Do we know if that helped? Curious that the indexer might have taken more than 10s to index a bit of new data.

Comment (dominikl) [reply components/tools/OmeroJava/test/integration/gateway/SearchFacilityTest.java:74]: I could set it to 10s again and see what happens?

Comment (mtbc) [reply components/tools/OmeroJava/test/integration/gateway/SearchFacilityTest.java:74]: I'll open a PR for 5s to really test it. :smiley:

Comment (mtbc) [reply components/tools/OmeroJava/test/integration/gateway/SearchFacilityTest.java:74]: Now opened #5959.

**Pull request merged**